packages/medusa/src/strategies/admin-authentication.ts

```diff
@@ -0,0 +1,175 @@
+            )
+            if (success) {
+              return done(null, user)
+            } else {
+              return done("Incorrect Username / Password")
+            }
+          } catch (error) {
+            return done(error)
+          }
+        }
+      )
+    )
+
+    // After a user has authenticated a JWT will be placed on a cookie, all
+    // calls will be authenticated based on the JWT
+    const { jwt_secret } = configModule.projectConfig
+    passport.use(
+      new JWTStrategy(
+        {
+          jwtFromRequest: (req): string => req.session.jwt,
+          secretOrKey: jwt_secret,
+        },
+        async (jwtPayload, done) => {
+          return done(null, jwtPayload)
+        }
+      )
+    )
+
+    // Alternatively use bearer token to authenticate to the admin api
+    passport.use(
+      new BearerStrategy(async (token, done) => {
+        const auth = await authService.authenticateAPIToken(token)
+        if (auth.success) {
+          done(null, auth.user)
+        } else {
+          done(auth.error)
+        }
+      })
+    )
+
+    app.use(passport.initialize())
+    app.use(passport.session())
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async shouldUseStrategy(
+    req: Request,
+    scope: "admin" | "store"
+  ): Promise<boolean> {
+    return scope === "admin"
+  }
+
+  async authenticate(req: Request, res: Response): Promise<void> {
+    const { jwt_secret } = this.configModule_.projectConfig
+    if (!jwt_secret) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        "Please configure jwt_secret in your environment"
+      )
+    }
+    const validated = await validator(AdminPostAuthReq, req.body)
+
+    const authService: AuthService = req.scope.resolve("authService")
+    const result = await authService.authenticate(
+      validated.email,
+      validated.password
+    )
+
+    if (result.success && result.user) {
+      // Add JWT to cookie
+      req.session.jwt = jwt.sign({ userId: result.user.id }, jwt_secret, {
+        expiresIn: "24h",
+      })
+
+      const cleanRes = _.omit(result.user, ["password_hash"])
+
+      res.json({ user: cleanRes })
+    } else {
+      res.sendStatus(401)
+    }
+  }
+
+  async unAuthenticate(req: Request, res: Response): Promise<void> {
+    req.session.destroy()
+    res.status(200).end()
+  }
+
+  async validate(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    passport.authenticate(["jwt", "bearer"], { session: false })(req, res, next)
+  }
+
+  private static validateConfig(configModule: ConfigModule): void | never {
+    const isProduction = ["production", "prod"].includes(
+      process.env.NODE_ENV || ""
+    )
+    const errorHandler = isProduction
+      ? (msg: string): never => {
+          throw new Error(msg)
+        }
+      : console.log
+
+    const jwt_secret =
+      configModule?.projectConfig?.jwt_secret ?? process.env.JWT_SECRET
```

Comment:
you added this to the config loader a couple of months back, we can just to `configmodule.projectconfig.jwt_secret`

Reply:
totaly 🚀